File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        
        res = 0
        vowelCnt = 0
        left = 0
        vowels = "aeiou"

        for right in range(len(s)):
            if s[right] in vowels:            
                vowelCnt += 1

            if right >= k and s[right-k] in vowels:
                    vowelCnt -=1

            res = max(res, vowelCnt)           

        return res

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        
        int res = 0;
        int left = 0;
        int vowelCount = 0;

        Set<Character> vowels = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u'));

        for(int right=0; right<s.length(); right++){
            if (vowels.contains(s.charAt(right))) vowelCount += 1;

            if (right >= k && vowels.contains(s.charAt(right-k))) vowelCount -= 1;

            res = Math.max(res, vowelCount);
        }

        return res;
    }
}